rfcs/036-ml-feature-discovery.md

```diff
@@ -0,0 +1,195 @@
+- Feature Name: ml_feature_discovery
+- Start Date: 2021-05-05
+- RFC PR: [amundsen-io/rfcs#36](https://github.com/amundsen-io/rfcs/pull/36) (after opening the RFC PR, update this with a link to it and update the file name)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# ML Features Discovery in Amundsen
+
+## Summary
+
+Amundsen currently indexes a few first-class entities, however we currently do not index any ML features that are easily discoverable for users. The goal of this work is to index ML features at the same level as Tables, Dashboards, and People resources in Amundsen.
+
+## Motivation
+
+Integration of ML projects in Amundsen has been heavily requested by the community and has already been implemented by other similar projects. This RFC outlines what will be the first step towards indexing ML projects in Amundsen since feature engineering is one of the most time consuming processes for ML engineers. Providing them with the ability to easily discover features that they can reuse for their models will significatly increase their productivity.
+
+## Guide-level Explanation (aka Product Details)
+
+### New Metadata Definitions
+#### Semantic Metadata
+Metadata related to the feature itself.
+- Key: what makes a feature uniquely identifiable is the group, name and version. We don’t use db/source because the same feature can exists in different places (like hive and dynamo)
+    - ex: test_feature_group_name/test_feature_name/1.2.0
+- Availability/sources: list of places where feature is available
```

Comment:
could you describe the scenario on why the same feature could be available in different places(hive, dynamo, etc)? are they generated by different pipelines(streaming,batch) to different sinks with the same IDL schema?

Reply:
We do have cases where we generate offline features using hive/presto etc. The same feature then also gets loaded into dynamodb.

So the same feature end up in multiple stores, and depending on the context ML model execution may choose the store (low latency use dynamodb, high latency is acceptable but high throughput is needed use hive) that is more appropriate. For example ride_count_30_days as a feature may get used by an online model for fraud detection which has low latency requirement and uses dynamodb. The exact same feature also gets used in batch model predictions for weekly coupon distribution which uses hive (or presto) for feature retrieval.